### ISSUE TITLE
get_geometric_layout results in missing required arguments for get_layout_for_multiple_components()

### ISSUE THREAD

**Issue by white-gecko** [paulbrodersen/netgraph#100]
In a jupyter notebook I plot the graph first …

```
import matplotlib.pyplot as plt

from netgraph import Graph # pip install netgraph OR conda install -c conda-forge netgraph

# right triangle
edge_length = {
    (0, 1) : 1.41,
    (0, 2) : 2.77,
    (2, 3) : 2.46,
    (0, 4) : 2.43,
    (1, 2) : 1.79,
    (1, 3) : 1.55,
    (3, 4) : 1.07,
}
edges = list(edge_length.keys())

fig, ax = plt.subplots()
Graph(edges, edge_labels=edge_length, node_layout='geometric',
      node_layout_kwargs=dict(edge_length=edge_length), ax=ax, scale=(3, 3), tol=0.000000001)
ax.set_aspect('equal')
plt.show()
```
… and then try to get the positions

```
from netgraph import get_geometric_layout

pos = get_geometric_layout(edges, edge_length)
```

This results in the following TypeError

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[63], line 3
      1 from netgraph import get_geometric_layout
----> 3 pos = get_geometric_layout(edges, edge_length)

File ~/.cache/pypoetry/virtualenvs/vermessung-2oHJ2gDh-py3.12/lib/python3.12/site-packages/netgraph/_node_layout.py:67, in _handle_multiple_components.<locals>.wrapped_layout_function(edges, nodes, *args, **kwargs)
     64         return get_layout_for_multiple_components(
     65             edges, components, layout_function, mode='side-by-side', *args, **kwargs)
     66     else:
---> 67         return get_layout_for_multiple_components(
     68             edges, components, layout_function, mode='packed', *args, **kwargs)
     69 else:
     70     return layout_function(edges, *args, **kwargs)

TypeError: get_layout_for_multiple_components() missing 2 required positional arguments: 'origin' and 'scale'
```

**Comment by paulbrodersen** [comment]
This one is on me. Most layout algorithms have a mandatory `origin` and `scale` parameter, corresponding to the (x, y)-coordinates in bottom-left corner of the canvas and its (width, height), respectively. These parameters are consumed by a function decorator that handles the arrangement of multiple components (most layout algorithms are only valid for single component graphs, so the arrangement of the components has to be handled separately), which is why the arguments are not documented by the automatically generated doc-strings.  

Tl;dr: this should do the trick:

```python
pos = get_geometric_layout(edges, edge_length, origin=(0,0), scale=(1,1))
```

**Comment by white-gecko** [comment]
That was the obvious thing, I have tried, but resulted in:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 3
      1 from netgraph import get_geometric_layout
----> 3 pos = get_geometric_layout(edges, edge_length, origin=(0,0), scale=(1,1))

File ~/.cache/pypoetry/virtualenvs/vermessung-2oHJ2gDh-py3.12/lib/python3.12/site-packages/netgraph/_node_layout.py:67, in _handle_multiple_components.<locals>.wrapped_layout_function(edges, nodes, *args, **kwargs)
     64         return get_layout_for_multiple_components(
     65             edges, components, layout_function, mode='side-by-side', *args, **kwargs)
     66     else:
---> 67         return get_layout_for_multiple_components(
     68             edges, components, layout_function, mode='packed', *args, **kwargs)
     69 else:
     70     return layout_function(edges, *args, **kwargs)

File ~/.cache/pypoetry/virtualenvs/vermessung-2oHJ2gDh-py3.12/lib/python3.12/site-packages/netgraph/_node_layout.py:126, in get_layout_for_multiple_components(edges, components, layout_function, origin, scale, mode, *args, **kwargs)
    124 if len(component) > 1:
    125     subgraph = _get_subgraph(edges, component)
--> 126     component_node_positions = layout_function(subgraph, origin=bbox[:2], scale=bbox[2:], *args, **kwargs)
    127     node_positions.update(component_node_positions)
    128 else:
    129     # component is a single node, which we can simply place at the centre of the bounding box

TypeError: get_geometric_layout() missing 1 required positional argument: 'edge_length'
```

**Comment by paulbrodersen** [comment]
Note to self: rework doc-strings for decorated functions, in particular in `_node_layout.py`

**Comment by white-gecko** [comment]
Thank you for this hint, with:

```
from netgraph import get_geometric_layout

pos = get_geometric_layout(edges, edge_length=edge_length, scale=(3, 3))
```

It works, while `scale` only is required for my dimensions, greater than 1.

You can decide, if you want to keep this issue open ;-)

**Comment by paulbrodersen** [comment]
Yeah, then I won't forget to fix the doc-strings.
